src/Misc/layoutroot/run.cmd

```diff
@@ -24,10 +24,10 @@ if /i "%~1" equ "localRun" (
   rem ********************************************************************************
   "%~dp0bin\Runner.Listener.exe" run %*
 
-  rem Return code 4 means the run once runner received an update message.
+  rem Return codes 3 and 4 mean the runner received an update message.
   rem Sleep 5 seconds to wait for the update process finish and run the runner again.
-  if ERRORLEVEL 4 (
+  if ERRORLEVEL 3 (
     timeout /t 5 /nobreak > NUL
-    "%~dp0bin\Runner.Listener.exe" run %*
+    "%~dp0run.cmd" %*
```

Comment:
if in some cases, this script restarts the runner (i.e. when "run once" runner) then it looks like this change is required if the runner were to receive a second update

(although i wonder if a `goto` would make more sense, e.g. to effectively create a loop)

Reply:
otherwise if you dont use `goto` then you end up with a bunch of parent cmd.exe processes after a bunch of updates.

Reply:
I'm certainly not a CMD guru, so I might well be wrong, but I thought calling another CMD file resulted in that file "taking over" the current process. In my testing I've at least observed that the output from the updated runner shows up in the same command prompt window. Happy to use `goto` if that is preferred though!

Reply:
ok cool i think you might be right. i forgot the default behavior isn't CALL. instead the script interpreter executes the other script never returns :)

side note, iirc we can't update the cmd file because the cmd.exe interpreter only loads a line at a time and we can't replace the cmd file while it's being executed. So calling the cmd file again does not enable us to use an updated cmd file. Disclaimer, i havent thought about these details in years, so i might be misremembering. @TingluoHuang might remember offhand whether i'm wrong :)